src/overlays/actors/ovl_En_Ani/z_en_ani.c

```diff
@@ -21,39 +45,311 @@ const ActorInit En_Ani_InitVars = {
+    this->actor.sub_98.mass = 0xFF;
+    if (LINK_IS_CHILD) {
+        EnAni_SetupAction(this, func_809B064C);
+    } else {
+        EnAni_SetupAction(this, func_809B07F8);
+    }
+    this->unk_2AA = 0;
+    this->unk_2A8 = 0;
+    this->actor.minVelocityY = -1.0f;
+    this->actor.velocity.y = -1.0f;
+}
+
+void EnAni_Destroy(EnAni* this, GlobalContext* globalCtx) {
+    ColliderCylinderMain* collider;
+    collider = &this->collider;
+    ActorCollider_FreeCylinder(globalCtx, &this->collider);
+}
+
+s32 EnAni_SetText(EnAni* this, GlobalContext* globalCtx, u16 textId) {
+    this->actor.textId = textId;
+    this->unk_2A8 |= 1;
+    func_8002F2CC(&this->actor, globalCtx, 100.0f);
+    return 0;
+}
+
+void func_809B04F0(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx) != 0) {
+        EnAni_SetupAction(this, func_809B064C);
+    }
+}
+
+void func_809B0524(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F334(&this->actor, globalCtx) != 0) {
+        EnAni_SetupAction(this, func_809B07F8);
+    }
+}
+
+void func_809B0558(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F410(&this->actor, globalCtx) != 0) {
+        this->actor.attachedA = 0;
+        if (LINK_IS_CHILD) {
+            EnAni_SetupAction(this, func_809B04F0);
+        } else {
+            EnAni_SetupAction(this, func_809B0524);
+        }
+        gSaveContext.item_get_inf[1] = gSaveContext.item_get_inf[1] | 0x20;
+        return;
+    }
+    func_8002F434(&this->actor, globalCtx, GI_HEART_PIECE, 10000.0f, 200.0f);
+}
+
+void func_809B05F0(EnAni* this, GlobalContext* globalCtx) {
+    if (func_8002F334(this, globalCtx) != 0) {
+        EnAni_SetupAction(this, func_809B0558);
+    }
+    func_8002F434(this, globalCtx, GI_HEART_PIECE, 10000.0f, 200.0f);
+}
+
+void func_809B064C(EnAni* this, GlobalContext* globalCtx) {
+    u16 textId;
+    s16 rotDiff;
+    u16 textId2;
+
+    textId2 = Text_GetFaceReaction(globalCtx, 0xA);
+    textId = textId2 & 0xffff;
+
+    if (!textId) {}
+
+    if (textId2 == 0) {
+        textId = gSaveContext.night_flag ? 0x5051 : 0x5050; // "...all I can do is look at the stars."
+                                                            // "...all I can do is look at Death Mountain."
+    }
+
+    rotDiff = this->actor.rotTowardsLinkY - this->actor.shape.rot.y;
+    if (func_8002F194(&this->actor, globalCtx) != 0) {
+        if (this->actor.textId == 0x5056) { // "To get a good view..."
+            EnAni_SetupAction(this, func_809B04F0);
+        } else if (this->actor.textId == 0x5055) { // "...I'll give you this as a memento."
+            EnAni_SetupAction(this, func_809B05F0);
+        } else {
+            EnAni_SetupAction(this, func_809B04F0);
+        }
+    } else if (rotDiff >= -0x36AF && rotDiff < 0 && this->actor.xzDistanceFromLink < 150.0f 
+                                                    && -80.0f < this->actor.yDistanceFromLink) {
```

Comment:
I dont believe this is out style for multiline conditions. Did clang do this? If so then disregard. If not, run it through clang